inc/saiswitch.h

```diff
@@ -406,6 +406,140 @@ typedef enum _sai_tunnel_decap_ecn_mode_t
 
 } sai_tunnel_decap_ecn_mode_t;
 
+/**
+ * @brief SAI switch soft error recovery error type.
+ */
+typedef enum _sai_switch_soft_error_type_t
+{
+    /**
+     * @brief Unknown error type
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_UNKNOWN = 0,
+
+    /**
+     * @brief Parity error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_PARITY = 1,
+
+    /**
+     * @brief ECC single bit error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ECC_SINGLE_BIT = 2,
+
+    /**
+     * @brief ECC double bit error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ECC_DOUBLE_BIT = 3,
+
+    /**
+     * @brief All soft error types
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ALL = 4
+} sai_switch_soft_error_type_t;
+
+/**
+ * @brief SAI switch soft error recovery error correction type.
+ */
+typedef enum _sai_switch_correction_type_t
+{
+    /**
+     * @brief S/W takes no action when error
+     * happens (Like some working memories)
+     */
+    SAI_SWITCH_CORRECTION_TYPE_NO_ACTION = 0,
+
+    /**
+     * @brief S/W tries to correct error but fails
+     */
+    SAI_SWITCH_CORRECTION_TYPE_FAIL_TO_CORRECT = 1,
+
+    /**
+     * @brief S/W writes NULL entry to clear the error
+     */
+    SAI_SWITCH_CORRECTION_TYPE_ENTRY_CLEAR = 2,
+
+    /**
+     * @brief Restore entry from a valid S/W cache
+     */
+    SAI_SWITCH_CORRECTION_TYPE_CACHE_RESTORE = 3,
+
+    /**
+     * @brief Restore entry from another pipe
+     */
+    SAI_SWITCH_CORRECTION_TYPE_HW_CACHE_RESTORE = 4,
+
+    /**
+     * @brief Memory needs special correction handling
+     */
+    SAI_SWITCH_CORRECTION_TYPE_SPECIAL = 5,
+} sai_switch_correction_type_t;
+
+/**
+ * @brief Soft error recovery flag information
+ *
+ * Note enum values must be powers of 2 to be used as bit mask
+ *
+ * @flags Contains flags
+ */
+typedef enum _sai_switch_ser_log_t
+{
+    /**
+     * @brief Error happens on memory
+     */
+    SAI_SWITCH_SER_LOG_MEM = 1 << 0,
+
+    /**
+     * @brief Error happens on register
+     */
+    SAI_SWITCH_SER_LOG_REG = 1 << 1,
+
+    /**
+     * @brief Parity errors detected more than once
+     */
+    SAI_SWITCH_SER_LOG_MULTI = 1 << 2,
+
+    /**
+     * @brief Error be corrected by S/W
+     */
+    SAI_SWITCH_SER_LOG_CORRECTED = 1 << 3,
+
+    /**
+     * @brief Corrupt memory entry data is provided in log
+     */
+    SAI_SWITCH_SER_LOG_ENTRY = 1 << 4,
+
+    /**
+     * @brief Cache data is valid
+     */
+    SAI_SWITCH_SER_LOG_CACHE = 1 << 5,
+} sai_switch_ser_log_t;
+
+/**
+ * @brief SAI switch soft error recovery info.
+ */
+typedef struct _sai_switch_ser_info_t
+{
+    /**
+     * @brief Error detected time
```

Comment:
maybe a little bit more information is this seconds or milliseconds or usec, and if usec, maybe this should be uint64 ?